ovn/utilities/ovn-nbctl.c

```diff
@@ -2810,6 +2817,204 @@ normalize_prefix_str(const char *orig_prefix)
     }
 }
 
+
+static void
+nbctl_lr_policy_add(struct ctl_context *ctx)
+{
+    const struct nbrec_logical_router *lr;
+
+    char *error = lr_by_name_or_uuid(ctx, ctx->argv[1], true, &lr);
+    if (error) {
+        ctx->error = error;
+        return;
+    }
+
+    if (ctx->argc >= 5) {
+       VLOG_INFO("nbctl_lr_policy_add: priority %s match %s action %s",
+         ctx->argv[2], ctx->argv[3], ctx->argv[4]);
+    } else {
+       ctl_fatal("insufficient parameters");
+    }
+
+    int64_t priority = parse_priority(ctx->argv[2]);
+    const char *action = ctx->argv[4];
+    char *next_hop = NULL;
+
+    /* Validate action. */
+    if (strcmp(action, "allow") && strcmp(action, "drop")
+        && strcmp(action, "reroute")) {
+        ctl_fatal("%s: action must be one of \"allow\", \"drop\", "
+                  "and \"reroute\"", action);
+    }
+
+    if (!strcmp(action, "reroute")) {
+        if (ctx->argc < 6) {
+            ctl_fatal("Nexthop is not specified when action is reroute.");
+        }
+    }
+
+    /* Check if same routing policy already exists. 
+     * A policy is uniquely identified by priority and match */
+    for (int i = 0; i < lr->n_policies; i++) {
+        const struct nbrec_logical_router_policy *policy = lr->policies[i];
+        if ((policy->priority == priority) &&
+            (!strcmp(policy->match, ctx->argv[3]))) {
+           ctl_fatal("Same routing policy already existed on the "
+                       " logical router %s.", ctx->argv[1]);
+        }
+    }
+
+    struct nbrec_logical_router_policy *policy;
+    policy = nbrec_logical_router_policy_insert(ctx->txn);
+    nbrec_logical_router_policy_set_priority(policy, priority);
+    nbrec_logical_router_policy_set_match(policy, ctx->argv[3]);
+    nbrec_logical_router_policy_set_action(policy, action);
+    if (ctx->argc == 6) {
+        next_hop = normalize_prefix_str(ctx->argv[5]);
+        if (!next_hop) {
+          ctl_fatal("bad next hop argument: %s", ctx->argv[5]);
+        }
+        nbrec_logical_router_policy_set_nexthop(policy, next_hop);
+    }
+
+    nbrec_logical_router_verify_policies(lr);
+    struct nbrec_logical_router_policy **new_policies
+        = xmalloc(sizeof *new_policies * (lr->n_policies + 1));
+    memcpy(new_policies, lr->policies,
+               sizeof *new_policies * lr->n_policies);
+    new_policies[lr->n_policies] = policy;
+    nbrec_logical_router_set_policies(lr, new_policies,
+                                           lr->n_policies + 1);
+    free(new_policies);
+    if (next_hop != NULL)
+        free(next_hop);
+}
+
+static void
+nbctl_lr_policy_del(struct ctl_context *ctx)
+{
+    const struct nbrec_logical_router *lr;
+    char *error = lr_by_name_or_uuid(ctx, ctx->argv[1], true, &lr);
+    if (error) {
+        ctx->error = error;
+        return;
+    }
+
+    if (ctx->argc == 2) {
+        /* If a priority is not specified, delete all policies. */
+        nbrec_logical_router_set_policies(lr, NULL, 0);
```

Comment:
Most likely, i am missing something.
For deletion, we are just removing the reference from nbrec_logical_router->policies.
Whereas in addition, we create a new row in nbrec_logical_router_policy.

Reply:
Implementation same as acls and static route.